docs/Package.md

```diff
@@ -123,6 +125,98 @@ add_action(
 ```
 
+## Building the package
+
+Sometimes, especially in unit tests, it might be desirable to obtain services as defined for the
+production code, but without calling any `ExecutableModule::run()`, which usually contains
+WP-dependant code, and therefore requires heavy mocking.
+
+For example, assuming a common `plugin()` function like the following:
+
+```php
+function plugin(): Modularity\Package {
+    static $package;
+    if (!$package) {
+        $properties = Modularity\Properties\PluginProperties::new(__FILE__);
+        $package = Modularity\Package::new($properties)
+            ->addModule(new ModuleOne())
+            ->addModule(new ModuleTwo())
+    }
+    return $package;
+}
+```
+
+In unit test it will be possible (as of v1.7+) to do something like the following:
+
+```php
+$myService = plugin()->build()->container()->get(MyService::class);
+static::assertTrue($myService->isValid());
+```
+
+### Booting a built container
+
+The `Package::boot()` method can be called on already built package.
+
+For example, the following is a valid unit test code:
+
+```php
+$plugin = plugin()->build();
+$myService = $plugin->container()->get(MyService::class);
+
+static::assertTrue($myService->isValid());
+static::assertFalse($myService->isBooted());
+
+$plugin->boot();
+
+static::assertTrue($myService->isBooted());
+```
+
+### Deprecated boot parameters
+
+Before Modularity v1.7.0, it was an accepted practice to pass default modules to `Package::boot()`,
+as in:
+
+```php
+add_action(
+	'plugins_loaded',
+	static function(): void {
+		plugin()->boot(new ModuleOne(), new ModuleTwo());
+	}
+);
+```
+
+This is now deprecated to allow a better separation of the "building" and "booting" steps.
+
+While it still works (and it will work up to version 2.0), it will emit a deprecation notice.
+
+The replacement is using `Package::addModule()` or the v1.7+ `Package::build()`:
+
+```php
+// Option 1:
+plugin()->addModule(new ModuleOne())->addModule(new ModuleTwo())->boot();
+
+// Option 2 (v1.7+ only):
+plugin()->build(new ModuleOne(), new ModuleTwo())->boot();
+```
+
+There's only one case in which calling `Package::boot()` with default modules will throw an 
+exception (besides triggering a deprecated notice), that is when a passed modules was not added
+before (neither via `Package::addModule()` nor `Package::build()`) and an instance of the container
+was already obtained from the package.
+
+For example, this will throw an exception:
+
+```php
+$plugin = plugin()->build();
+
+// Now that container is built, passing modules to `boot()` will raise an exception, because we
+// can't add new modules to an already "compiled" container being that read-only.
+$container = $plugin->container();
+
+$plugin->boot(new ModuleOne());
+```
+
+Either calling `$plugin->boot()` _before_ `container()`, or passing `new ModuleOne()` to the `build()` call, will prevent the exception to be thrown, but the deprecation will still be emitted.
 
 
 ## Connecting packages
```

Comment:
Should we also update the "What happens on `Package::boot()` section? https://github.com/inpsyde/modularity/blob/c5242ec64469f189d32f5c32f9c4a2769b9455b7/docs/Package.md#what-happens-on-packageboot

Reply:
That part of the documentation could be improved, yes.